source/linux/background-managers/hyprpaper.js

```diff
@@ -0,0 +1,91 @@
+import childProcess from 'node:child_process';
+import {commandExists, execFile} from '../util.js';
+
+export async function isAvailable() {
+	return commandExists('hyprpaper');
+}
+
+async function initialize() {
+	return new Promise((resolve, reject) => {
+		const hyprpaper = childProcess.spawn('script', ['--quiet', '--command', 'hyprpaper', '/dev/null']);
+		hyprpaper.stdout.on('data', data => {
+			if (data.toString().includes('[ERR]') || data.toString().includes('error')) {
```

Comment:
This looks really fragile. Would be better to check exit code.

Reply:
Then where it should resolve?

Reply:
In the subprocess exit handler.

Reply:
But hyprpaper must always work in background

Reply:
I assume it exits if it encounters an error?

Reply:
Yes. But I think at error we should reject promise not resolve.

Reply:
I never said it should not reject.